src/adapter/src/error.rs

```diff
@@ -443,6 +449,18 @@ impl fmt::Display for AdapterError {
             AdapterError::UncallableFunction { func, context } => {
                 write!(f, "cannot call {} in {}", func, context)
             }
+            AdapterError::UnallowedOnCluster {
+                depends_on,
+                cluster,
+            } => {
+                let items = depends_on.into_iter().map(|item| item.quoted()).join(", ");
+                write!(
+                    f,
+                    "querying the following items {items} is not allowed from the {} cluster. \
+                    Use `SET CLUSTER = <cluster-name>` to change your cluster and re-run the query.",
```

Comment:
This second sentence should probably be a hint or detail. See https://github.com/MaterializeInc/materialize/blob/main/doc/developer/style.md#error-message-style

Reply:
Thanks!